src/Interval.hs

```diff
@@ -167,46 +167,46 @@ intervalToDistance int@(Interval q i) =
     subIntervalToDistance (Interval IMajor 6) = Just 9
     subIntervalToDistance (Interval IMajor 7) = Just 11
 
-    subIntervalToDistance (Interval IMinor i) = 
-      let defQuality = defaultIQuality i 
-      in 
-        case defQuality of 
-          IMajor -> (subtract 1) <$> subIntervalToDistance (Interval IMajor i)
+    subIntervalToDistance (Interval IMinor i) =
+      let defQuality = defaultIQuality i
+      in
+        case defQuality of
+          IMajor -> subtract 1 <$> subIntervalToDistance (Interval IMajor i)
           IPerfect -> Nothing
           _ -> error "Impl error, default quality must be M or P"
 
 
-    subIntervalToDistance (Interval (IAugmented x) i) = 
+    subIntervalToDistance (Interval (IAugmented x) i) =
       let defQuality = defaultIQuality i
-      in 
-        ((+) x) <$> subIntervalToDistance (Interval defQuality i)
+      in
+        (x +) <$> subIntervalToDistance (Interval defQuality i)
```

Comment:
Don't forget that binary operators have special syntax for partial application (sections.)

Reply:
I should be a little more detailed here: Binary (two-argument) functions have some special things you can do with them.

 First, _any_ function whose type is of the form `a -> b -> c` (that is, a binary function) can be used as an infix operator by surrounding the function's name with backquotes (shiftless tilde key). So, for example, if we have:

```haskell
elem :: a -> Set a -> Bool
```

That can be used in two ways given something of the element type `x` and a set `s`:

```haskell
-- The 'normal' way
elem x s

-- The infix way
x `elem` s
```

Second, infix functions (and, of course, symbolic operators like `+`, `$`, etc) can be partially applied using a special syntax called _sections_. Let `@@` stand for some arbitrary infix operator. First, we note that the expression

```haskell
(@@)
```

 Allows us to use `@@` as if it were a regularly named function, i.e. in prefix notation. So, 

```haskell
(@@) x y == x @@ y
```

For all pairs of valid inputs. Naturally, you may ask "do parentheses undo backquotes around a named function?" And the answer is yes. If `op` is a binary function, then

```haskell
(`op`) x y == op x y
```

Again for all pairs of valid inputs.

Sections are an extension to this syntax: We surround the operator with parentheses, but we can put an argument to either side of the operator to partially apply it to that value _in the position implied by the syntactic structure of the section_. To make that precise, here's roughly what sections de-sugar to. As before, let `@@` stand for some arbitrary infix operator:

```haskell
-- The 'left' case is easy, since we can just apply the function as normal
(e @@) == (@@) e

-- The 'right' case needs a slight change, but nothing too fancy
(@@ e) == flip (@@) e
```

Where `flip` is the function of type `(a -> b -> c) -> (b -> a -> c)`.